src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -40,6 +46,9 @@ internal sealed class SqlClientDiagnosticListener : ListenerHandler
     private readonly PropertyFetcher<Exception> exceptionFetcher = new("Exception");
     private readonly SqlClientInstrumentationOptions options;
 
+#if NET6_0_OR_GREATER
+    [RequiresUnreferencedCode(SqlClientInstrumentation.SqlClientTrimmingUnsupportedMessage)]
```

Comment:
You don't need this on both the class and the constructor. You can remove this on the constructor.